commander/src/utils/reader.ts

```diff
@@ -236,14 +236,20 @@ const getNestedPropertyTemplate = (schema: Schema): NestedPropertyTemplate => {
 };
 
 const castValue = (
-	strVal: string,
+	val: string,
 	schemaType: string,
-): number | string | Record<string, unknown> => {
+): number | bigint | string | string[] | Record<string, unknown> => {
 	if (schemaType === 'object') {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
-		return JSON.parse(strVal);
+		return JSON.parse(val);
 	}
-	return Number.isInteger(Number(strVal)) ? Number(strVal) : strVal;
+	if (schemaType === 'array') {
+		return val.split(',');
+	}
+	if (Number.isInteger(Number(val))) {
```

Comment:
What if the `val` here is a an integer string but schemaType suggest that dataType should be string? I suggest to check `schemaType` not that value is integer or not.